ddl/ddl_worker.go

```diff
@@ -451,6 +465,19 @@ func newMetaWithQueueTp(txn kv.Transaction, tp string) *meta.Meta {
 	return meta.NewMeta(txn)
 }
 
+func (w *worker) setDDLLabelForTopSQL(job *model.Job) {
+	if !variable.TopSQLEnabled() || job == nil {
+		return
+	}
+
+	if job.Query != w.cacheSQL {
+		w.cacheNormalizedSQL, w.cacheDigest = parser.NormalizeDigest(job.Query)
+		w.cacheSQL = job.Query
+	}
+
+	w.ddlJobCtx = topsql.AttachSQLInfo(context.Background(), w.cacheNormalizedSQL, w.cacheDigest, "", nil, false)
```

Comment:
why we use context.Background as input and assign the result to w.ddlJobCtx?

Reply:
`w.ddlJobCtx` actually only use to execute internal SQL, since we should reset the DDL SQL label after finish execute internal SQL.
Before this PR, TiDB also uses `context.Background()` to execute internal SQL, so I use `context.Background` here to avoid change the original behaivor.

Reply:
since the worker's ddlJobCtx has been initiated as context.Background(), can it be like this?
w.ddlJobCtx = topsql.AttachSQLInfo(w.ddlJobCtx, w.cacheNormalizedSQL, w.cacheDigest, "", nil, false)